include/minisketch.h

```diff
@@ -151,6 +151,156 @@ size_t minisketch_compute_max_elements(uint32_t bits, size_t capacity, uint32_t
+
+    /** See minisketch_compute_capacity(). */
+    static size_t ComputeCapacity(uint32_t bits, size_t max_elements, uint32_t fpbits) noexcept { return minisketch_compute_capacity(bits, max_elements, fpbits); }
+
+    /** See minisketch_compute_max_elements(). */
+    static size_t ComputeMaxElements(uint32_t bits, size_t capacity, uint32_t fpbits) noexcept { return minisketch_compute_max_elements(bits, capacity, fpbits); }
+
+    /** Construct a clone of the specified sketch. */
+    Minisketch(const Minisketch& sketch) noexcept
+    {
+        m_minisketch = std::unique_ptr<minisketch, Deleter>(minisketch_clone(sketch.m_minisketch.get()));
+    }
+
+    /** Make this Minisketch a clone of the specified one. */
+    Minisketch& operator=(const Minisketch& sketch) noexcept
+    {
+        m_minisketch = std::unique_ptr<minisketch, Deleter>(minisketch_clone(sketch.m_minisketch.get()));
+        return *this;
+    }
+
+    explicit operator bool() const noexcept { return bool{m_minisketch}; }
+
+    Minisketch() noexcept = default;
+    Minisketch(Minisketch&&) noexcept = default;
+    Minisketch& operator=(Minisketch&&) noexcept = default;
+
+    /** Construct a Minisketch object with the specified parameters. */
+    Minisketch(uint32_t bits, uint32_t implementation, size_t capacity) noexcept
+    {
+        m_minisketch = std::unique_ptr<minisketch, Deleter>(minisketch_create(bits, implementation, capacity));
+    }
+
+    /** Create a Minisketch object sufficiently large for the specified number of elements at given fpbits. */
+    static Minisketch CreateFP(uint32_t bits, uint32_t implementation, size_t max_elements, uint32_t fpbits) noexcept
+    {
+        return Minisketch(bits, implementation, ComputeCapacity(bits, max_elements, fpbits));
+    }
+
+    /** See minisketch_get_bits(). */
+    uint32_t GetBits() const noexcept { return minisketch_bits(m_minisketch.get()); }
+
+    /** See minisketch_get_capacity(). */
+    size_t GetCapacity() const noexcept { return minisketch_capacity(m_minisketch.get()); }
+
+    /** See minisketch_get_implementation(). */
+    uint32_t GetImplementation() const noexcept { return minisketch_implementation(m_minisketch.get()); }
+
+    /** See minisketch_set_seed(). */
+    Minisketch& SetSeed(uint64_t seed) noexcept
+    {
+        minisketch_set_seed(m_minisketch.get(), seed);
+        return *this;
+    }
+
+    /** See miniksetch_add_element(). */
+    Minisketch& Add(uint64_t element) noexcept
+    {
+        minisketch_add_uint64(m_minisketch.get(), element);
+        return *this;
+    }
+
+    /** See minisketch_merge(). */
+    Minisketch& Merge(const Minisketch& sketch) noexcept
+    {
+        minisketch_merge(m_minisketch.get(), sketch.m_minisketch.get());
+        return *this;
+    }
+
+    /** Decode this sketch into the result vector, which must have a size equal to the maximum allowed elements. */
```

Comment:
> must have a size equal to the maximum allowed elements

This doesn't seem exactly true. I think `minisketch_decode` will return -1 if `result.size()` is too small, but it's allowed to be larger (and will be shrunk at the end by `result.resize()`. Perhaps the comment should be "must have a size at least as large as the maximum allowed elements"?

For my own understanding: is this requirement to avoid having to reallocate the vector, which would mean this would no longer be noexcept?

Reply:
> This doesn't seem exactly true. I think minisketch_decode will return -1 if result.size() is too small, but it's allowed to be larger (and will be shrunk at the end by result.resize()

Yes, that's correct. But I don't see how that's in conflict with the comment.

> Perhaps the comment should be "must have a size at least as large as the maximum allowed elements"?

No, that'd be wrong. If you make it larger than the number of elements you want, you may get more.

> For my own understanding: is this requirement to avoid having to reallocate the vector, which would mean this would no longer be noexcept?

No, just to avoid redundancy. The size of what you want is already passed by the size of the vector you provide, no need to provide it a second time.

Reply:
Oh, I think I see your confusion. It is intended to mean "You should pass in a vector with a size equal to the maximum number of elements *you the caller* allow it to return". "Allowed" is maybe not the best term here?

Reply:
Sorry, yes I was confused by who was allowing who! I'd usually expect the interface to document what the client is allowed to do.

I don't have a great suggestion for changing this. Perhaps something like "Decode up to result.size() elements from the sketch. Will fail if result is not large enough to hold all sketch elements." but I'm not sure if that captures everything.

Reply:
I've changed the comments a bit.